Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -938,40 +938,21 @@ size_t GroupDetectors2::formGroups(API::MatrixWorkspace_const_sptr inputWS,
     // are assumed to be the same here
     outSpec.setSharedX(inputWS->sharedX(0));
 
-    // the Y values and errors from spectra being grouped are combined in the
-    // output spectrum
-    auto &firstY = outSpec.mutableY();
     // Keep track of number of detectors required for masking
     size_t nonMaskedSpectra(0);
-    beh->mutableX(outIndex)[0] = 0.0;
-    beh->mutableE(outIndex)[0] = 0.0;
+
     for (auto originalWI : it->second) {
       // detectors to add to firstSpecNum
-      const auto &fromSpectrum = inputWS->getSpectrum(originalWI);
+      const auto &inputSpectrum = inputWS->getSpectrum(originalWI);
 
-      // Add up all the Y spectra and store the result in the first one
+      outSpec.setHistogram(outSpec.histogram() += inputSpectrum.histogram());
```

Comment:
Doh - A case of premature optimisation as in my head I was eliminating a variable.